storage/badger_storage.go

```diff
@@ -197,6 +197,7 @@ func NewBadgerStorage(
 		closed:        make(chan struct{}),
 		pool:          NewBufferPool(),
 		compress:      true,
+		writer:        utils.NewMutexMap(utils.DefaultShards),
```

Comment:
I'm going to make this configurable...usually would recommend making this equal to the write concurrency.